### ISSUE TITLE
Localized numbers

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#11114]
In Lua scripts you might want to output a number with decimal point like “5.34”.
Usuallly you will use `string.format` for this but the number will always be written in the English notation (decimal point), but other languages use other characters, like a comma. “5.34” in English is “5,34” in German.

I therefore request to make it somehow possible to localize numbers whenever they are user-facing. The minimum requirement is that the decimal point is converted, since that's the most important thing.

## Analysis

The preferred way to do this is by using `setlocale` in C++ or `os.setlocale` in Lua. However, this option is sadly not possible right now, because of this (in `gettext.cpp`):

```
         /* no matter what locale is used we need number format to be "C" */
         /* to ensure formspec parameters are evaluated correct!          */
 
         setlocale(LC_NUMERIC, "C");
```

The solution would be fairly easy and straight-forward if we use `setlocale` to set to the system locale (with `""`). Sadly, it turns out that the formspecs *depend* on the locale to be `"C"`. Apparently a locale-depending function is used to parse the numbers in formspecs. I tried to switch to my German system locale, but the main menu was completely broken.

This makes solving the problem much trickier.

So I wonder, how could the problem be solved?

### Ideas

My ideas:
* Somehow make all the formspec stuff locale-indepenent; also check if there are other things in the core (or builtin) which depend on the locale and make them independent from it, and then switch the numeric locale to `""`
* Add separate utility functions to generate localized numbers, keep the `"C"` locale

Maybe there's an even better approach …

**Comment by SmallJoker** [comment]
No, please don't. This will only cause confusion when example commands are shared across different locales. You would also change the C locale of the server; not the one of the client, hence missing the entire point.

:-1: not worth the effort.

**Comment by Wuzzy2** [comment]
The syntax of server commands is a special case and the locale can and should be completely ignored for them. I do !!!NOT!!! want to Excel-ify Minetest. :-) I consider server commands to be a language in itselves (like a programming language) and I absolutely do not wish to localize the syntax of them in any way. I am aware what a terrible mess this would create. That's why I am not proposing it.

I actually only care about UI stuff. In places where you just print out numbers in player-facing texts in the UI (like "5.3 kg wheat" or "1.23 seconds") it is totally justified to localize them.

I am aware this isn't easy to solve, unfortunately. If I would have an easy solution to this, I would already have posted a PR. ;-)

**Comment by rubenwardy** [comment]
This makes sense if you add number formatting to client-side translations. It doesn't make sense for string.format or anything that effects command parsing

**Comment by Wuzzy2** [comment]
This was closed unjustifiably IMO. The problem this issue shows, namely, that you cannot have localized numbers in mods (e.g. "5,4 Kilogramm Fleisch" (=5.4 kilos of meat)). We only have `string.format` but it only spits out numbers in "English" notation (with decimal point).

If you write "5.4 Kilogramm Fleisch" in German, that's a grammtical *error*.

Sadly, I don't know yet HOW to solve this problem without creating a huge mess. Maybe there is a simpler solution, I just haven't figured it out yet … :-(
